lib/grape-swagger/doc_methods.rb

```diff
@@ -47,6 +47,12 @@ def parse_params(params, path, method)
                           'double'
                         when 'Symbol'
                           'string'
+                        when /^\[(?<type>.*)\]$/
+                          items[:type] = Regexp.last_match[:type].downcase
+                          if PRIMITIVE_MAPPINGS.key?(items[:type])
+                            items[:type], items[:format] = PRIMITIVE_MAPPINGS[items[:type]]
```

Comment:
Later we deal with parsed_params[:type], not items[:type], items just affected into parsed_params[:items].